lib/src/builder/build_step.dart

```diff
@@ -13,6 +13,12 @@ abstract class BuildStep {
   /// The primary input for this build step.
   Asset get input;
 
+  /// Checks if an [Asset] by [id] exists as an input for this [BuildStep].
+  ///
+  /// If [trackAsDependency] is true, then [id] will be marked as a dependency
+  /// of all [expectedOutputs].
+  Future<bool> hasInput(AssetId id, {bool trackAsDependency: true});
```

Comment:
What does it mean if the user passes *false* for this? Presumable, the caller uses the boolean result of this method to do *something* that affects their build output. If that boolean were to change without the build step being re-run (because it passed `trackAsDependency: false`), won't you end up with stale or inconsistent output?